FogFog-iOS/FogFog-iOS/OAuth/AppleOAuthService.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  AppleOAuthService.swift
+//  FogFog-iOS
+//
+//  Created by EUNJU on 2023/07/04.
+//
+
+import UIKit
+
+import AuthenticationServices
+import RxSwift
+
+final class AppleOAuthService: OAuthServiceType {
+    
+    private let disposeBag = DisposeBag()
+    private let appleLoginManager = AppleLoginManager()
+    
+    func authorize() -> Single<OAuthAuthentication> {
+        return login().map { OAuthAuthentication(oauthType: .apple, idToken: $0.id, code: $0.code)}
+    }
+    
+    private func login() -> Single<AppleUser> {
+        return Single.create { observer in
+            self.appleLoginManager
+                .handleAuthorizationAppleIDButtonPress()
+                .subscribe(onNext: { result in
+                    guard let auth = result.credential as? ASAuthorizationAppleIDCredential else { return }
```

Comment:
p3;
auth에서 옵셔널 체크가 되겠지만, 아래와 같이 작성해도 괜찮지 않을까 싶어요!
```suggestion
                    guard 
                      let auth = result.credential as? ASAuthorizationAppleIDCredential,
                      let idToken = auth.identityToken, 
                      let code = auth.authorizationCode 
                    else { return }
```